modules/planning/pipeline/feature_generator.cc

```diff
@@ -491,6 +496,11 @@ void FeatureGenerator::GenerateRoutingFeature(
   }
 
   if (routing_index < 0) {
+    std::ostringstream msg;
+    msg << "no LOCAL routing. frame_num["
+        << learning_data_frame->frame_num() << "]";
```

Comment:
Try `StrCat`.

Reply:
if performance is not a concern(well, it is not ,here), stream seems better here. It IS a logging output with  flexible text. 
please approve. :)